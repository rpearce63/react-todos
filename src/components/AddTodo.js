import React, { Component } from "react";
import { connect } from "react-redux";
import { addTodo } from "../actions";

export class AddTodo extends Component {
  state = { todo: "" };

  addTodo = () => {
    this.props.addTodo(this.state.todo);
    this.setState({ todo: "" });
  };
  render() {
    return (
      <div className="add_todo">
        <input
          className="add_todo--input"
          type="text"
          value={this.state.todo}
          onChange={e => this.setState({ todo: e.target.value })}
        />
        <button
          className="add_todo--button"
          onClick={this.addTodo}
          disabled={!this.state.todo && "disabled"}
        >
          Add Todo
        </button>
      </div>
    );
  }
}

export default connect(
  null,
  { addTodo }
)(AddTodo);
