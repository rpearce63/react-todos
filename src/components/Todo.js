import React, { Component } from "react";
import { connect } from "react-redux";
import * as actions from "../actions";

export class Todo extends Component {
  handleOnChange = () => {
    this.props.updateStatus(this.props.todo.id);
  };
  removeTodo = () => {
    if (window.confirm("Are you sure?")) {
      this.props.removeTodo(this.props.todo.id);
    }
  };

  render() {
    return (
      <div className="todo_item">
        <input
          className="todo_item--checkbox"
          type="checkbox"
          checked={this.props.todo.complete}
          onChange={this.handleOnChange}
        />
        <div className="todo_item--text">{this.props.todo.text}</div>
        <div className="todo_item--remove" onClick={this.removeTodo}>
          {" "}
          x{" "}
        </div>
      </div>
    );
  }
}

export default connect(
  null,
  actions
)(Todo);
