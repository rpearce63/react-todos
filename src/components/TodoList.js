import React, { Component } from "react";
import Todo from "./Todo";
import { connect } from "react-redux";
import * as actions from "../actions";

export class TodoList extends Component {
  componentDidMount() {
    //this.props.fetchTodos();
  }
  render() {
    //console.log("render with ", this.props.todos);
    return (
      <div>
        <h3>Todos</h3>
        {this.props.todos.map(todo => (
          <Todo key={todo.id} todo={todo} />
        ))}
      </div>
    );
  }
}

export default connect(
  state => ({ todos: state.todos }),
  actions
)(TodoList);
