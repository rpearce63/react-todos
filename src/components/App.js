import React, { Component } from "react";
import AddTodo from "./AddTodo";
import TodoList from "./TodoList";
import "../styles/index.css";

export default class App extends Component {
  render() {
    return (
      <div>
        <h1>My Todos</h1>
        <hr />
        <AddTodo />
        <TodoList />
      </div>
    );
  }
}
