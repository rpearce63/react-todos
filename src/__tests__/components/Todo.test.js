import React from "react";
import { shallow } from "enzyme";

import { Todo } from "../../components/Todo";

describe("Todo component", () => {
  const props = {
    todo: {
      id: 1,
      text: "a todo",
      complete: false
    },
    updateStatus: jest.fn(),
    removeTodo: jest.fn()
  };
  const wrapper = shallow(<Todo {...props} />);

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the todo values", () => {
    expect(wrapper.find("#todo_text").text()).toEqual("a todo");
    expect(wrapper.find("input").prop("checked")).toBe(false);
  });

  it("should dispatch UPDATE_STATUS action", () => {
    wrapper.find("input").simulate("change");
    expect(props.updateStatus).toHaveBeenCalledWith(props.todo.id);
  });

  it("should not dispatch remove if no selected on confirmation", () => {
    window.confirm = jest.fn(() => false);
    wrapper.find("#todo_remove").simulate("click");
    expect(props.removeTodo).not.toHaveBeenCalled();
  });

  it("should dispatch REMOVE_TODO action", () => {
    window.confirm = jest.fn(() => true);
    wrapper.find("#todo_remove").simulate("click");
    expect(window.confirm).toBeCalled();
    expect(props.removeTodo).toHaveBeenCalledWith(props.todo.id);
  });
});
