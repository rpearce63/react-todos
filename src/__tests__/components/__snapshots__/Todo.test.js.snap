// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Todo
    removeTodo={[MockFunction]}
    todo={
      Object {
        "complete": false,
        "id": 1,
        "text": "a todo",
      }
    }
    updateStatus={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />,
        <div
          id="todo_text"
        >
          a todo
        </div>,
        <div
          id="todo_remove"
          onClick={[Function]}
        >
           
          x
           
        </div>,
      ],
      "style": Object {
        "display": "flex",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": false,
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "a todo",
          "id": "todo_text",
        },
        "ref": null,
        "rendered": "a todo",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "x",
            " ",
          ],
          "id": "todo_remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "x",
          " ",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />,
          <div
            id="todo_text"
          >
            a todo
          </div>,
          <div
            id="todo_remove"
            onClick={[Function]}
          >
             
            x
             
          </div>,
        ],
        "style": Object {
          "display": "flex",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "a todo",
            "id": "todo_text",
          },
          "ref": null,
          "rendered": "a todo",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "x",
              " ",
            ],
            "id": "todo_remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "x",
            " ",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
